builtin/providers/google/resource_compute_region_instance_group_manager.go

```diff
@@ -0,0 +1,420 @@
+
+			managedInstanceCount := len(managedInstances.ManagedInstances)
+			instances := make([]string, managedInstanceCount)
+			for i, v := range managedInstances.ManagedInstances {
+				instances[i] = v.Instance
+			}
+
+			recreateInstances := &compute.RegionInstanceGroupManagersRecreateRequest{
+				Instances: instances,
+			}
+
+			op, err = config.clientCompute.RegionInstanceGroupManagers.RecreateInstances(
+				project, region, d.Id(), recreateInstances).Do()
+
+			if err != nil {
+				return fmt.Errorf("Error restarting instance group managers instances: %s", err)
+			}
+
+			err = computeOperationWaitRegionTime(config, op, project, region,
+				managedInstanceCount*4, "Restarting RegionInstanceGroupManagers instances")
+			if err != nil {
+				return err
+			}
+		}
+
+		d.SetPartial("instance_template")
+	}
+
+	if d.HasChange("named_port") {
+		namedPorts := getNamedPorts(d.Get("named_port").([]interface{}))
+		setNamedPorts := &compute.RegionInstanceGroupsSetNamedPortsRequest{
+			NamedPorts: namedPorts,
+		}
+
+		op, err := config.clientCompute.RegionInstanceGroups.SetNamedPorts(
+			project, region, d.Id(), setNamedPorts).Do()
+		if err != nil {
+			return fmt.Errorf("Error updating RegionInstanceGroupManager: %s", err)
+		}
+
+		err = computeOperationWaitRegion(config, op, project, region, "Updating RegionInstanceGroupManager")
+		if err != nil {
+			return err
+		}
+
+		d.SetPartial("named_port")
+	}
+
+	if d.HasChange("target_size") {
+		if v, ok := d.GetOk("target_size"); ok {
+			targetSize := int64(v.(int))
+
+			op, err := config.clientCompute.RegionInstanceGroupManagers.Resize(
+				project, region, d.Id(), targetSize).Do()
+			if err != nil {
+				return fmt.Errorf("Error updating RegionInstanceGroupManager: %s", err)
+			}
+
+			err = computeOperationWaitRegion(config, op, project, region, "Updating RegionInstanceGroupManager")
+			if err != nil {
+				return err
+			}
+		}
+
+		d.SetPartial("target_size")
+	}
+
+	d.Partial(false)
+
+	return resourceComputeRegionInstanceGroupManagerRead(d, meta)
+}
+
+func resourceComputeRegionInstanceGroupManagerDelete(d *schema.ResourceData, meta interface{}) error {
+	config := meta.(*Config)
+
+	project, err := getProject(d, config)
+	if err != nil {
+		return err
+	}
+
+	region, err := getRegion(d, config)
+	if err != nil {
+		return err
+	}
+
+	var op *compute.Operation
+	for i := 0; i < 40; i++ {
+		op, err = config.clientCompute.RegionInstanceGroupManagers.Delete(project, region, d.Id()).Do()
+		if err != nil {
+			time.Sleep(4000 * time.Millisecond)
+			continue
+		}
+	}
+	if gerr, ok := err.(*googleapi.Error); ok {
+		if gerr.Code == 404 {
+			log.Printf("[WARN] Removing Instance Group Manager %q because it's gone", d.Get("name").(string))
+
+			d.SetId("")
+
+			err = nil
+		} else {
+			return fmt.Errorf("Error deleting region instance group manager: %s", err)
+		}
+	}
+
+	currentSize := int64(d.Get("target_size").(int))
+
+	for err != nil && currentSize > 0 {
```

Comment:
Can we comment this to better describe what's going on? Why are we backing off on error? Is it possible to use a [`StateChangeConf`](https://godoc.org/github.com/hashicorp/terraform/helper/resource#StateChangeConf) to wait for this instead of custom backoff code?